extensions/amp-a4a/0.1/a4a-variable-source.js

```diff
@@ -113,10 +112,56 @@ export class A4AVariableSource extends VariableSource {
       return getNavigationData(this.win_, 'redirectCount');
     });
 
+    this.set('HTML_ATTR',
+        /** @type {function(...*)} */(this.htmlAttrBinding_.bind(this)));
+
     for (let v = 0; v < WHITELISTED_VARIABLES.length; v++) {
       const varName = WHITELISTED_VARIABLES[v];
       const resolvers = this.globalVariableSource_.get(varName);
       this.set(varName, resolvers.sync).setAsync(varName, resolvers.async);
     }
   }
+
+  /**
+   * Provides a binding for getting attributes from the DOM.
+   * Most such bindings are provided in src/service/url-replacements-impl, but
+   * this one needs access to this.win_.document, which if the amp-analytics
+   * tag is contained within an amp-ad tag will NOT be the parent/publisher
+   * page. Hence the need to put it here.
+   * @param {string} cssSelector Elements matching this selector will be
+   *     included, provided they have at least one of the attributeNames
+   *     set, up to a max of 10. May be URI encoded.
+   * @param attributeNames The attributes whose values will be returned.
+   * @returns {string}
+   */
+  htmlAttrBinding_(cssSelector, ...attributeNames) {
```

Comment:
@lannka I moved this to here as you requested, but it seems that initialize() on line 90 above is never getting called, so the variable substitution never occurs. Do you know why this is?

Reply:
as shown in `variable-source.js` L147, it's lazily called. Can you double check?

Reply:
Right, variable-source.js:147 calls that. But to test, I put:
```
console.log('Getting: ' + name);
```
right there and ran it with the example included in this CL, and it never got called. I'm still looking into why, but figured you might know right away.
Double-checked that this still runs, at amp-a4a.js:1389:
```
          installUrlReplacementsForEmbed(this.getAmpDoc(), embedWin,
              new A4AVariableSource(this.getAmpDoc(), embedWin));
```
It seems like this is being caused by this line in variables.js:
```
      const raw = options.vars[name] != null ? options.vars[name] : '';
```
In this case, name = 'HTML_ATTR' but options.vars does not contain name. It seems that options.vars is set to the value of default.vars in vendors.js. This means that we ultimately end up in encodeVars(), with raw="", and that method then just returns "".
So, I propose re-adding HTML_ATTR in vendors.js.
But even after re-adding that, when we get to expandUrlAsync() in url-replacements-impl.js, url is correct but the other args to that method are empty.

Reply:
I will wait here until I get more direction from you, @lannka, to avoid going down some long rabbit hole that maybe you can solve in 30 seconds :)

Reply:
@lannka Any thoughts here? Just looking for a yes/no/don't know as to whether I'm proceeding on the right path. Thanks!

Reply:
I patched your PR locally and did a quick debug. Seems like the other comment I put your other PR fixes this:

```
this.urlReplacementService_ = Services.urlReplacementsForDoc(ampElement.element);
```